client/main_test.go

```diff
@@ -79,27 +81,35 @@ func createCodecs() {
 }
 
 func createStreamIdGenerators() {
-	var managed = func(int) int16 {
+	var managed = func(clientId int, version primitive.ProtocolVersion) int16 {
 		return client.ManagedStreamId
 	}
-	var fixed = func(clientId int) int16 {
+	var fixed = func(clientId int, version primitive.ProtocolVersion) int16 {
 		if int16(clientId) == client.ManagedStreamId {
 			panic("stream id 0")
 		}
 		return int16(clientId)
 	}
-	counter := uint32(0)
-	var incremental = func(clientId int) int16 {
+	counter := uint32(1)
+	var incremental = func(clientId int, version primitive.ProtocolVersion) int16 {
```

Comment:
`clientId` isn't used anymore? Also, why don't you use a buffered channel that is pre-populated with the stream ids?